shared/providers.ts

```diff
@@ -343,8 +363,13 @@ export function badgeValues<T extends object>(
 }
 
 /**
- * Converts an async generator provider into an observable provider. This also
- * memoizes the previous result as a workaround for #1321 (below).
+ * Converts an async generator provider into an observable provider. This
+ * also memoizes the previous result. This reduces the number of definition
+ * requests from two to one on each hover.
+ *
+ * The memoization was originally a workaround for #1321 (below), but should
```

Comment:
I didn't think of this when you made the original change. My only concern about  memoization is that it's potentially an unbounded memory leak. What's the dynamic extent of the wrapper? Is it coeval with the process or some smaller transaction?

Not a blocker here (the change was already made) but I wanted to bring it up because I'm concerned that it could be hard to fix later.